internal/pkg/v2/application/validator/metrics/metrics_test.go

```diff
@@ -0,0 +1,83 @@
+/*******************************************************************************
+ * Copyright 2020 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package metrics
+
+import (
+	"testing"
+
+	"github.com/edgexfoundry/edgex-go/internal/pkg/v2/application"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/v2/application/delegate"
+	dtoBase "github.com/edgexfoundry/edgex-go/internal/pkg/v2/application/dto/v2dot0/base"
+	dtoError "github.com/edgexfoundry/edgex-go/internal/pkg/v2/application/dto/v2dot0/error"
+	dto "github.com/edgexfoundry/edgex-go/internal/pkg/v2/application/dto/v2dot0/metrics"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/v2/infrastructure/test"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// TestValidate tests the Validator for Version request.
+func TestValidate(t *testing.T) {
+	readingID := test.FactoryRandomString()
+	invalidRequestType := "string is not the request type we're expecting."
+
+	variations := []struct {
+		name             string
+		request          interface{}
+		expectedResponse interface{}
+		expectedStatus   application.Status
+	}{
+		{
+			name:             "valid",
+			request:          dto.NewRequest(dtoBase.NewRequest(readingID)),
+			expectedResponse: dto.NewRequest(dtoBase.NewRequest(readingID)),
```

Comment:
You really did review this code in detail!

This is a validator test.  It leverages `delegate.TestExecutable` to wrap the validation; if all validation passes, `delegate.TestExecutable.Execute()` returns the original request.  We test for responses in the other test cases because we expect an error response to be returned.  Here, we're not actually taking any action if everything validates -- we just check that what is returned (a request) is what we sent (a request).